src/internal/connector/onedrive/collections.go

```diff
@@ -434,15 +466,62 @@ func (c *Collections) UpdateCollections(
 			// already created and partially populated.
 			updatePath(newPaths, *item.GetId(), folderPath.String())
 
+			if ok {
+				// If we find prev path, that implies these are
+				// moved/renames folders and not new ones
+				prevPath, err := path.FromDataLayerPath(prevColPath, true) // FIXME(meain): converting string to path
+				if err != nil {
+					logger.Ctx(ctx).Errorw("invalid prev path for item", "error", err)
+					return err
+				}
+
+				col, found := c.CollectionMap[folderPath.String()]
+				if !found {
+					col := NewCollection(
+						c.itemClient,
+						folderPath,
+						prevPath,
+						driveID,
+						c.service,
+						c.statusUpdater,
+						c.source,
+						c.ctrl,
+						invalidPrevDelta,
+					)
+					c.CollectionMap[folderPath.String()] = col
+					c.NumContainers++
```

Comment:
we only want to make new collections if the folder was moved. Otherwise we risk backup up empty folders which we've avoided thus far in OneDrive (not sure if we should rethink that at some point)
```suggestion
					if prevPath != nil {
						col := NewCollection(
							c.itemClient,
							folderPath,
							prevPath,
							driveID,
							c.service,
							c.statusUpdater,
							c.source,
							c.ctrl,
							invalidPrevDelta,
						)
						c.CollectionMap[folderPath.String()] = col
						c.NumContainers++
					}
```